### PR TITLE
Updates for 0.12

### PR DESCRIPTION
**Pull request by kritzcreek**
I think we just shouldn't be using purescript-argonaut-codecs anymore, but I just made the update for 0.12 and didn't change anything else. I didn't update purescript-psa and I don't really plan on doing that, but I thought I'd push this anyway in case it's useful to anyone else.

### CODE REVIEW

File: src/Psa/Output.purs [hunk 1]
```diff
@@ -222,22 +223,20 @@ trimPosition lines pos =
...
 
   trimCol col l =
-    case Str.charAt (col - 2) l of
+    case Str.codePointAt (col - 2) l of
       Just x | isPunc x -> trimCol (col - 1) l
       Just _            -> trimComment col l
       _                 -> Nothing
 
-  -- TODO: this breaks if "--" is inside a quoted string.
+  -- TODO: this breaks if "--" is insidentitye a quoted string.
```
Comment (natefaubion) [inline src/Psa/Output.purs:231]: Heh.

File: src/Psa/Printer.purs [hunk 2]
```diff
@@ -53,12 +55,12 @@ instance monoidRow :: Monoid (Row a) where
 
 data AnsiText
   = Plain String
-  | Style (Array Ansi.GraphicsParam) String
+  | Style (NonEmptyList Ansi.GraphicsParam) String
```
Comment (natefaubion) [inline src/Psa/Printer.purs:58]: What was the reason for the `NonEmptyList` change? Also why not `NonEmptyArray`?

Comment (kritzcreek) [reply src/Psa/Printer.purs:58]: I think purescript-ansi was changed? I was also a bit confused, but I just rolled with it.

File: src/Psa/Output.purs (hunk 1 above)
Comment (kritzcreek) [reply src/Psa/Output.purs:231]: whoops :D

**Pull request merged**